### PR TITLE
Added new column  to sys.shards

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/reference/sys/shard/blob/BlobShardMinLuceneVersionExpression.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import io.crate.metadata.ReferenceImplementation;
+import io.crate.operation.reference.sys.shard.ShardInfoExtractor;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.index.shard.IndexShard;
+
+public class BlobShardMinLuceneVersionExpression implements ReferenceImplementation<BytesRef> {
+
+    private IndexShard indexShard;
+
+    public BlobShardMinLuceneVersionExpression(IndexShard indexShard) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/reference/sys/shard/blob/BlobShardMinLuceneVersionExpression.java]: Only one implementation is required which can be used for blob and non-blob

File: sql/src/main/java/io/crate/operation/reference/sys/shard/ShardInfoExtractor.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+package io.crate.operation.reference.sys.shard;
+
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.index.engine.EngineClosedException;
+import org.elasticsearch.index.shard.IndexShard;
+
+public final class ShardInfoExtractor {
+
+    public static BytesRef retrieveMinLuceneVersion(IndexShard indexShard) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/reference/sys/shard/ShardInfoExtractor.java]: why is this in a separate class intead of putting it into the expression as well?

File: sql/src/main/java/io/crate/operation/reference/sys/shard/ShardMinLuceneVersionExpression.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    private IndexShard indexShard;
+
+    public ShardMinLuceneVersionExpression(IndexShard indexShard) {
+        this.indexShard = indexShard;
+    }
+
+    @Override
+    public BytesRef value() {
+        // We are not caching the value so if you do
+        // `select min_lucene_version, min_lucene_version, ... from sys.shards`
```
Comment (seut) [inline sql/src/main/java/io/crate/operation/reference/sys/shard/ShardMinLuceneVersionExpression.java]: could you pls add why we're not caching? (e.g. expression is a shard singleton and survives a single query)

File: CHANGES.txt [hunk 4]
```diff
@@ -5,6 +5,9 @@ Changes for Crate
 Unreleased
 ==========
 
+ - Added new column ``min_lucene_version`` which to ``sys.shards`` table,
```
Comment (seut) [inline CHANGES.txt]: remove `which`

**Pull request closed without merging**